modules/RTC/TraceablePeerConnection.js

```diff
@@ -2314,10 +2335,15 @@ TraceablePeerConnection.prototype.getStats = function(callback, errback) {
                 // Making sure that getStats won't fail if error callback is
                 // not passed.
             }));
-    } else if (browser.isSafariWithWebrtc()) {
-        // FIXME: Safari's native stats implementation is not compatibile with
-        // existing stats processing logic. Skip implementing stats for now to
-        // at least get native webrtc Safari available for use.
+    } else if (browser.isSafariWithWebrtc() || browser.isFirefox()) {
+        // uses the new Promise based getStats
+        this.peerconnection.getStats()
+            .then(callback)
+            .catch(errback || (() => {
+
+                // Making sure that getStats won't fail if error callback is
+                // not passed.
+            }));
```

Comment:
If this is promise based then maybe you should return this.peerconnection.getStats() so that in the future the callbacks can be removed. This would only be if nothing depends on TraceablePeerConnection#getStats not returning a promise.

Reply:
Currently, RTPStatsCollector implements both the promise and non-promise getStats. So we always call it passing callbacks. 
We can avoid passing callbacks, with very little code duplications here:
https://github.com/jitsi/lib-jitsi-meet/blob/e3adbca60887eacd964b8336fc477202e270117f/modules/statistics/RTPStatsCollector.js#L314
Do you want me to change it? But then TraceablePeerConnection.getStats in one case will depend on the callbacks and in the other case will just return Promise:
 * @returns {Promise|void}, is this ok?

Reply:
I guess we can wait if the promise isn't actually used yet. I was being too preemptive.